.github/workflows/ci.yml

```diff
@@ -158,6 +158,42 @@ jobs:
         with:
           name: "${{ matrix.IMAGE.TOXENV }} on ${{ matrix.IMAGE.IMAGE }}"
 
+  linux-arm64:
+    runs-on: [self-hosted, linux, ARM64]
+    container: ghcr.io/pyca/cryptography-runner-${{ matrix.IMAGE.IMAGE }}
+    strategy:
+      fail-fast: false
+      matrix:
+        IMAGE:
+          - {IMAGE: "ubuntu-focal:aarch64", TOXENV: "py38"}
+    name: "${{ matrix.IMAGE.TOXENV }} on ${{ matrix.IMAGE.IMAGE }}"
+    timeout-minutes: 20
```

Comment:
So nothing here is conditional on it being you or I :-) What's the security story with this running for everyone?

The answer is that anyone could run a malicious payload and escape the container. What do they get access to if they do that? Do they get a priviledged github token?

Reply:
They definitely get a GH token. It's not clear to me what permissions it has. Does `permissions: read-all` constrain it usefully? Can any user just change that as part of the PR? Do we consider that risk low given the workflow approval requirement for new users?

Reply:
The approval requirement only applies to brand new github accounts, so it's a very weak control.

I'm super nervous because I do not understand the security properties of this.

As an alternative idea, what do you think about making this a 2nd workflow file which only runs on `main`, just so we can see it working?

Reply:
I originally had it that way actually. I can also add github.actor restrictions (`if: github.actor == "alex" || github.actor == "reaperhulk"`), but that restriction is in the yaml itself so it seems highly likely that a user can just...remove it.